docs/introduction/index.md

```diff
@@ -0,0 +1,109 @@
+
+## Running .NET in the browser
+
+Running .NET code inside web browsers is made possible by a relatively new technology, [WebAssembly](http://webassembly.org) (abbreviated *wasm*). WebAssembly is an open web standard and is supported in web browsers without plugins. WebAssembly is a compact bytecode format optimized for fast download and maximum execution speed.
+
+WebAssembly code can access the full functionality of the browser via JavaScript interop. At the same time, WebAssembly code runs in the same trusted sandbox as JavaScript to prevent malicious actions on the client machine.
+
+When a Blazor app is built and run in a browser:
+
+1. C# code files and Razor files are compiled into .NET assemblies.
+1. The assemblies and the .NET runtime are downloaded to the browser.
+1. Blazor uses JavaScript to bootstrap the .NET runtime and configures the runtime to load required assembly references. Document object model (DOM) manipulation and browser API calls are handled by the Blazor runtime via JavaScript interoperability.
+
+## Browsers that don't support WebAssembly
+
+The .NET runtime is supplied as a WebAssembly binary and an [asm.js](https://wikipedia.org/wiki/Asm.js)-based implementation. *asm.js* is a subset of JavaScript and can be executed by JavaScript runtimes in browsers going back several years. When Blazor loads in the browser, it checks for WebAssembly support. If WebAssembly isn't supported, the *asm.js* runtime is loaded. *asm.js* isn't always used because it's larger and slower than the WebAssembly runtime.
+
+## Blazor components
+
+Blazor apps are built with *components*. A component is a piece of UI, such as a page, dialog, or data entry form. Components can be nested, reused, and shared between projects.
+
+In Blazor, a component is a .NET class. The class can either be written directly, as a C# class (*\*.cs*), or more commonly in the form of a Razor markup page (*\*.cshtml*).
+
+[Razor](https://docs.microsoft.com/aspnet/core/mvc/views/razor) is a syntax for combining HTML markup with C# code. Razor is designed for developer productivity, allowing the developer to switch between markup and C# in the same file with [IntelliSense](https://docs.microsoft.com/visualstudio/ide/using-intellisense) support. The following markup is an example of a basic custom dialog component in a Razor file (*DialogComponent.cshtml*):
+
+```cshtml
+<div>
+    <h2>@Title</h2>
+    @RenderContent(Body)
+    <button onclick=@OnOK>OK</button>
+</div>
+
+@functions {
+    public string Title { get; set; }
+    public Content Body { get; set; }
+    public Action OnOK { get; set; }
+}
+```
+
+When this component is used elsewhere in the app, IntelliSense speeds development with syntax completion and parameter info.
+
+Components can be:
+
+* Nested.
+* Generated procedurally in code.
```

Comment:
I'm not sure what it means to generate a component procedurally. Do you just mean by writing it as a `.cs` file instead of `.cshtml`? Because I'd regard that as a syntax difference rather than a conceptual difference.

Reply:
Yes, I don't think I stated that well. I did mean "in code." Let's see if this works better ...

> * Created with Razor (*\*.cshtml*) or C# (*\*.cs*) code.